File: Ejercicio1/FactoriaCarretera.java
package Ejercicio1;

import java.util.ArrayList;

public class FactoriaCarretera implements FactoriaCarreraYBicicleta{

    @Override
    public Bicicleta crearBicicleta() {
        return new BicicletaCarretera();
        
    }

    @Override
    public Carrera crearCarrera(ArrayList<Bicicleta> bicis) {
        return new CarreraCarretera(bicis);
    }

}

File: Ejercicio2/FactoriaCarretera.py
from .FactoriaCarreraYBicicleta import FactoriaCarreraYBicicleta
from .Carrera import Carrera
from .Bicicleta import Bicicleta
from .CarreraCarretera import CarreraCarretera
from .BicicletaCarretera import BicicletaCarretera
from typing import List

class FactoriaCarretera(FactoriaCarreraYBicicleta):
    def crear_bicicleta(self) -> Bicicleta:
        return BicicletaCarretera()

    def crear_carrera(self, bicis: List[Bicicleta]) -> Carrera:
        return CarreraCarretera(bicis)